LFS-935 & LFS-936: Publish a manifest file of available NeuralCR / BasicCR text analysis models and provide a downloader utility

IntegralProgrammer opened the pull request:
This pull request implements the tasks LFS-935 and LFS-936 by:

- Publishing a manifest file of available NeuralCR / BasicCR text analysis models
- Providing a utility program to download / verify / extract the published models so that they may be made available to CARDS

To test:
1. Build this branch
2. In a new directory, clone the https://github.com/IntegralProgrammer/NeuralCR repository and checkout the `LFS-935_936_test` branch
3. Build with `docker build -t ccmsk/neuralcr .`
4. Go back to the `lfs` (CARDS) repository and enter the `compose-cluster` directory
5. `python3 generate_compose_yaml.py --enable_ncr --oak_filesystem`
6. `mkdir NCR_MODEL`
7. Navigate to the `Utilities/NCR-Downloader` directory
8. `python3 download_model.py --download pmc_model_new.bin --savedir ../../compose-cluster/NCR_MODEL/`
9. `python3 download_model.py --download HP --savedir ../../compose-cluster/NCR_MODEL/`
10. Navigate back to the `compose-cluster` directory
11. Edit the `docker-compose.yml` file, adding the environment variable `ADDITIONAL_RUN_MODES=lfs`, to the configuration for the `lfsinitial` container (ie. right below `OAK_FILESYSTEM=true`)
12. `docker-compose build`
13. `docker-compose up -d`
14. Visit http://localhost:8080, login as admin, create a new _Demographics_ Form, test the notes annotation for the _Co-Morbidities_ section (you may want to have the *HP* vocabulary installed so that human-readable ontology names persist when the Form is closed).

IntegralProgrammer committed:
LFS-936: Create a NCR model downloader which will read a models manifest file, download the selected model, verify its integrity, and extract it to the appropriate directory

IntegralProgrammer committed:
LFS-935: A manifest file should be published which contains the URLs and cryptographic hashes for all available NeuralCR / BasicCR models

k-pace reviewed (approved):
I was able to follow the steps and have models downloaded for me. NCR text annotations worked, installation of HP linked to human readable names.

marta- commented:
Could the above instructions be turned into a bash script?

sdumitriu commented on Utilities/NCR-Downloader/download_model.py:
```diff
@@ -0,0 +1,100 @@
+    sys.exit(-1)
+
+  #Extract the .tar.gz file if necessary
+  if model_type in TAR_GZ_TYPES:
```
Why not check the file extension itself?

IntegralProgrammer replied:
Good idea - I have implemented this with commit `4c4dd5a` :heavy_check_mark:

sdumitriu reviewed (commented):
Running without any arguments, this just prints "Done", while nothing gets actually done. Should we require that at least one of `help, list, download` is specified, and if not print a message other than "Done"?

IntegralProgrammer committed:
LFS-936: Create a NCR model downloader which will read a models manifest file, download the selected model, verify its integrity, and extract it to the appropriate directory

If neither --list nor --download are specified, print an error message stating that these arguments are mandatory

IntegralProgrammer committed:
LFS-936: Create a NCR model downloader which will read a models manifest file, download the selected model, verify its integrity, and extract it to the appropriate directory

Determine if the downloaded model needs to be uncompressed based on the file extension provided in the model URL specified in the manifest file

IntegralProgrammer commented:
> Running without any arguments, this just prints "Done", while nothing gets actually done. Should we require that at least one of `help, list, download` is specified, and if not print a message other than "Done"?

Implemented by commit `54e6814` :heavy_check_mark:
